ethereum/rpc/namespaces/eth/api.go

```diff
@@ -604,7 +604,47 @@ func (e *PublicAPI) GetTransactionByHash(hash common.Hash) (*rpctypes.RPCTransac
 	res, err := e.GetTxByEthHash(hash)
 	if err != nil {
 		e.logger.WithError(err).Debugln("tx not found", "hash", hash.Hex())
-		return nil, nil
+
+		// try to find tx in mempool
+		txs, err := e.backend.PendingTransactions()
+		if err != nil {
+			return nil, nil
+		}
+
+		for _, tx := range txs {
+			if len((*tx).GetMsgs()) != 1 {
+				e.logger.Debugln("invalid tx")
+				return nil, fmt.Errorf("invalid tx type: %T", tx)
+			}
+			msg, ok := (*tx).GetMsgs()[0].(*evmtypes.MsgEthereumTx)
+			if !ok {
+				e.logger.Debugln("invalid tx")
+				return nil, fmt.Errorf("invalid tx type: %T", tx)
+			}
+
+			from, err := msg.GetSender(e.chainIDEpoch)
+			if err != nil {
+				return nil, err
+			}
+
+			data, err := evmtypes.UnpackTxData(msg.Data)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unpack tx data: %w", err)
+			}
+
+			rpctx, err := rpctypes.NewTransactionFromData(
+				data,
+				from,
+				msg.AsTransaction().Hash(),
+				common.Hash{},
+				uint64(0),
+				uint64(0),
+			)
+			if err != nil {
+				return nil, err
+			}
+			return rpctx, nil
```

Comment:
this is not filtering the tx by the hash

Reply:
good catch, fixed 😄